### DISCUSSION TITLE
Custom session object server side

### DISCUSSION THREAD

**Discussion by JasperAlexander** [nextauthjs/next-auth#7398] [Help] [answered]
How to make a custom session object that persists server side and can be accessed with getServerSession()? The docs state that the session object is not persisted server side and that only data such as the session token, the user, and the expiry time is stored in the session table. But why? How to store custom fields like id in the session table? Is there really no way to do this?

**Comment by ghost** [comment]
I use a database and it is stored, but you need to add the ID in the callback if you need to use it:
https://next-auth.js.org/configuration/callbacks#session-callback

If you use a JWT you also need to add it in this callback:
https://next-auth.js.org/configuration/callbacks#jwt-callback

**Comment by JasperAlexander** [comment, marked as answer]
I forgot to give authOptions as argument to getServerSession.
